### PR TITLE
Kylel/pipeline

### PR DESCRIPTION
**Pull request by kyleclo**

### CODE REVIEW

File: corvid/table_extraction/evaluate.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    """"""
+    if not is_same_dimensions(pred, gold):
+        raise Exception('Cant compare tables of different dimensions')
+
+    return agg([sim(gold[i, j], pred[i, j])
+                for i in range(gold.nrow)
+                for j in range(gold.ncol)]) / (gold.nrow * gold.ncol)
+
+
+def evaluate(gold_table: Table, pred_table: Table) -> Dict[str, float]:
```
Comment (kyleclo) [inline corvid/table_extraction/evaluate.py]: update design doc w/ this name

File: corvid/types/semantic_table.py [hunk 2]
```diff
@@ -1,19 +1,61 @@
...
\ No newline at end of file
+    VALID_CELL_CLASSES = ['LABEL', 'VALUE']
+
+    def __init__(self, table: Table):
+        self.cell_classes = self._classify_cells(table)
+        if self._should_transpose(table):
+            table = table.transpose()
+        self.table = self._normalize(table)
+
+    def _classify_cells(self, table: Table) -> List[List[str]]:
```
Comment (kyleclo) [inline corvid/types/semantic_table.py]: alternative implementations:
- insert `class-label` into each `cell` object
- wrap each `cell` object in a `SemanticCell` object & `SemanticTable` comrprised of `SemanticCells`

File: corvid/table_extraction/table_extractor.py [hunk 3]
```diff
@@ -151,6 +156,7 @@ def _create_table_from_tetml(cls, table_id: int, table_tag: Tag,
                     colspan=int(cell_tag.get('colspan')) \
                         if cell_tag.get('colspan') else 1
                 )
+                cell.bounding_box = cell.compute_bounding_box()
```
Comment (kyleclo) [inline corvid/table_extraction/table_extractor.py]: make property

File: corvid/table_extraction/table_extractor.py [hunk 4]
```diff
@@ -161,10 +167,12 @@ def _create_table_from_tetml(cls, table_id: int, table_tag: Tag,
...
-                      page_num=0,
-                      caption=caption)
+        table = Table.create_from_cells(
+            cells=cells,
+            nrow=len(ncol_per_row),
+            ncol=ncol_per_row[0],
+            paper_id='PAPER_ID',
+            page_num=0,
+            caption=caption)
+        table.bounding_box = table.compute_bounding_box()
```
Comment (kyleclo) [inline corvid/table_extraction/table_extractor.py]: make property

File: corvid/types/table.py [hunk 5]
```diff
@@ -36,20 +34,41 @@ class Cell(object):
...
+    def compute_bounding_box(self) -> Box:
+        """Finds bounding boxes that tightly bound all Tokens in the Cell"""
+        bounding_boxes = [token.bounding_box for token in self.tokens]
+
+        if len(bounding_boxes) == 0 and not all(bounding_boxes):
+            raise Exception('Tokens missing bounding boxes')
+
+        return Box.compute_bounding_box(bounding_boxes)
+
+    @classmethod
```
Comment (kyleclo) [inline corvid/types/table.py]: possibly place helper instantiators in a separate util for cell&tables

File: corvid/types/table.py [hunk 6]
```diff
@@ -162,13 +193,82 @@ def __str__(self):
...
-        new_cells = [
-            Cell(tokens=cell.tokens,
-                 rowspan=cell.colspan,
-                 colspan=cell.rowspan) for cell in self.cells
-        ]
-
-        return Table(cells=new_cells, nrow=self.ncol, ncol=self.nrow,
-                     paper_id=self.paper_id, page_num=self.page_num,
-                     caption=self.caption, is_row_wise=False)
+    def __eq__(self, other: 'Table') -> bool:
```
Comment (kyleclo) [inline corvid/types/table.py]: add comment that this compares only on cell values and not other members, like captions

File: main.py [hunk 7]
```diff
@@ -0,0 +1,130 @@
...
+        pdf_path = fetch_pdf(paper)
+        xml_path = parse_pdf(pdf_path)
+        tables.add(extract_tables(xml_path))
+
+    normalize(tables)
+    filter_relevant(tables)
+
+    gold_tables = load_gold_tables(dataset)
+    for gold_table in gold_tables:
+        target_schema = gold_table.schema
```
Comment (kyleclo) [inline main.py]: semantic table property

File: main.py [hunk 8]
```diff
@@ -0,0 +1,130 @@
...
+
+    # verify external dependencies
+    assert is_url_working(ES_PROD_URL)
+    assert is_url_working(S3_PDFS_URL)
+    assert os.path.exists(TET_BIN_PATH)
+
+    log_summary = {}
+
+    for dataset in datasets:
+        name, paper_id = dataset.get('name'), dataset.get('paper_id')
```
Comment (kyleclo) [inline main.py]: aliases

Comment (kyleclo) [inline main.py]: call `dataset_paper_id` and `dataset_name`

File: main.py [hunk 9]
```diff
@@ -0,0 +1,130 @@
...
+                log_summary[name]['parse_pdf_to_tetml']['success'] += 1
+            except Exception as e:
+                print(e)
+                log_summary[name]['parse_pdf_to_tetml']['fail'] += 1
+                continue
+
+            # extract tables from TETML
+            try:
+                with open(tetml_path, 'r') as f_tetml:
+                    tables = TetmlTableExtractor.extract_tables(
```
Comment (kyleclo) [inline main.py]: extend to `tables`

**Comment by cmkumar87** [comment]
Reviewed in person.  
- PR actually has lot more than the wrapping pipeline and has parts of the wrapper that remains to be implemented. eg. call to Table Filter . which is going to happen in the forthcoming PR. 
- Merging it at this helps minimal rebase to Baseline and other PRs for the methods. So, approving.

**Comment by kyleclo** [comment]
continue working on `main.py`

**Pull request merged**